### ISSUE TITLE
Purchase cost imported not the same as what indicated in the import template

### ISSUE THREAD

**Issue by ZechClar** [snipe/snipe-it#2388]
Purchase cost is not showing the same amount as indicated in the import template. No error received. Value of Purchase Cost imported is just not correct. 

SnipeIT version: v3.0-ebc74a65
OS: Windows Server 2012
Method used to install SnipeIT: Xampp

Picture below is the file I imported
![capture1](https://cloud.githubusercontent.com/assets/20469279/17359014/603fbfa4-5997-11e6-851c-1c672aa54272.JPG)

Picture below is what I see in SnipeIT
![capture2](https://cloud.githubusercontent.com/assets/20469279/17359015/6045b440-5997-11e6-86be-2c08338f7eb2.JPG)
#### Please confirm you have done the following before posting your bug report:
- [x] I have enabled debug mode
- [x] I have read [checked the Common Issues page](http://docs.snipeitapp.com/common-issues.html)

---
#### Please provide answers to these questions before posting your bug report:
- Version of Snipe-IT you're running
- What OS and web server you're running Snipe-IT on
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc)
- If you're getting an error in your browser, include that error
- What specific Snipe-IT page you're on, and what specific element you're interacting with to trigger the error
- If a stacktrace is provided in the error, include that too.
- Any errors that appear in your browser's error console.
- Confirm whether the error is [reproduceable on the demo](https://snipeitapp.com/demo).
- Include any additional information you can find in `app/storage/logs` and your webserver's logs.
- Include what you've done so far in the installation, and if you got any error messages along the way.
- Indicate whether or not you've manually edited any data directly in the database

**Comment by CLOZADA16** [comment]
Upgrade SnipeIT and tested the import. Same thing happens as what is shown in the screenshot I attached previously.

Snipe-IT version	v3.2.0-53-gfff2bce
PHP Version	5.6.23
Laravel Version	5.2.39

**Comment by snipe** [comment]
Can you check and make sure that in the CSV (plain text, not spreadsheet) that there are no commas in the cost?

**Comment by CLOZADA16** [comment]
no commas were added. All data are set as plain text.

if commas were included I would just receive an error "A non well formed numeric value encountered" when i run php artisan snipeit:import.

I'm also not able to use the import feature. It just remains on the bar you see on the image below. Got tired of waiting for 5-10min so I used the command line instead. I only have less than 200 items on the file i was importing.
![import](https://cloud.githubusercontent.com/assets/20469279/17386867/28b35a2a-5a21-11e6-885f-db2d577c6a71.JPG)

**Comment by dmeltzer** [comment]
Can you check the javascript debug console to see if it shows any errors on trying to upload?

The two causes I've seen of that are improper permissions on the storage/private_uploads/imports/assets directory or if your APP_URL isn't the same as the url you access from (Cross site javascript protection)

**Comment by CLOZADA16** [comment]
That fixes the import feature in the website. but the purchase cost is still incorrect. What it does is that if the value is less than a thousand (e.g. 134 or 500.35), the data imported will be correct. if the value is more than a hundred e.g. 1125, 45000.24, 125876 it removes the last 3 numbers including the decimal numbers showing it as 1, 45, 125 after import.

![import](https://cloud.githubusercontent.com/assets/20469279/17387236/5c8e81aa-5a24-11e6-8217-0de7483d245b.JPG)

**Comment by dmeltzer** [comment]
Yes, That's a separate issue.  Working on it though.

**Comment by dmeltzer** [comment]
I'm having trouble reproducing that issue.  Could you maybe post part of the csv you are trying to import? Just a few line with errors would be enough.  The following imports fine for me.

```
Name,Email,Username,item Name,item Category,Model name,Manufacturer,Model Number,Serial number,Asset Tag,Location,Notes,Purchase Date,Purchase Cost,Company,Status,Warranty,Supplier
Bonnie Nelson,bnelson0@cdbaby.com,bnelson0,eget nunc donec quis,quam,massa id,Linkbridge,6377018600094472,27aa8378-b0f4-4289-84a4-405da95c6147,970882174-8,Daping,"Curabitur in libero ut massa volutpat convallis. Morbi odio odio, elementum eu, interdum eu, tincidunt in, leo. Maecenas pulvinar lobortis est.",2016-04-05,$133289.59,Alpha,Undeployable,14,Blogspan
Judith Ferguson,jferguson1@state.tx.us,jferguson1,mi in porttitor,justo,congue diam id,Flipstorm,5.02043359569189E+018,4bc7fc90-5a97-412f-8eed-77ecacc643fc,544574073-0,Cirangga Kidul,,2016-03-08,76346,,Undeployable,12,Oyope
Mildred Gibson,mgibson2@wiley.com,mgibson2,morbi quis tortor id,nunc nisl duis,convallis tortor risus,Lajo,374622546776765,2837ab20-8f0d-4935-8a52-226392f2b1b0,710141467-2,Shekou,In congue. Etiam justo. Etiam pretium iaculis justo.,2015-08-09,233.57,Konklab,Lost,,
```

**Comment by dmeltzer** [comment]
I'm wondering if it's something weird excel does.  Or maybe a locale thing?  You could try inserting the following line after line 783 of app/Console/Commands/ObjectImportCommand.php and then see if you get more helpful information either from storage/logs/importer.log or running php artisan snipeit:import --verbose
```
            $this->log("Asset cost parsed: " . $asset->purchase_cost);
```

So That

```
        if (!empty($item["purchase_cost"])) {
            //TODO How to generalize this for not USD?
            $purchase_cost = substr($item["purchase_cost"], 0, 1) === '$' ? substr($item["purchase_cost"], 1) : $item["purchase_cost"];
            $asset->purchase_cost = number_format($purchase_cost, 2);
        } else {
```

changes to 

```
        if (!empty($item["purchase_cost"])) {
            //TODO How to generalize this for not USD?
            $purchase_cost = substr($item["purchase_cost"], 0, 1) === '$' ? substr($item["purchase_cost"], 1) : $item["purchase_cost"];
            $asset->purchase_cost = number_format($purchase_cost, 2);
            $this->log("Asset cost parsed: " . $asset->purchase_cost);
        } else {
```

**Comment by CLOZADA16** [comment]
I have the same code already in ObjectImportCommand.php since I upgraded it to v3.2. No changes made.

Sorry the excel screenshot looks small. Hope you can still read it.
![import](https://cloud.githubusercontent.com/assets/20469279/17389102/c2209832-5a33-11e6-932d-98a0dcbd5d47.JPG)
Here's the output after import
![import2](https://cloud.githubusercontent.com/assets/20469279/17389103/c2266f0a-5a33-11e6-8046-b21d1e0a42df.JPG)

**Comment by dmeltzer** [comment]
I need to see what the actual csv file looks like. I'm curious if it's
excel doing something special.  If I put those numbers into a plain csv
they import just fine, so we need to figure out what's different on your
end.

**Comment by CLOZADA16** [comment]
Name,Email,Username,Item Name,Category,Model Name,Manufacturer,Model Number ,Serial Number,Asset Tag,Location,Notes,Purchase Date,Purchase Cost,Company,Status,Warranty,Supplier
Asset Admin,admin@noemail.com,Administrator,IBM x3650 series as Servers purchase for the running of S,Server,System,IBM,x3650,06BKZB0,OIS22172,E5 - DC2 Server Rack 20,Asset Number: 10027087 / ,8-Aug-12,14117.23,RP-SOI,Ready to Deploy,,NULL
,,,IBM x3650 series as Servers purchase for the running of S,Server,System ,IBM,x3650,06BKZC5,OIS22173,E5 - DC2 Server Rack 20,Asset Number: 10027088 / ,8-Aug-12,114117,RP-SOI,Ready to Deploy,,NULL
,,,Ethernet Adapter for Server (IBM NetXtreme II 1000 PCI-E ),Ethernet Adapter,NetXtreme ,IBM,II PCI-E,X61S8TD,OIS24012,E5 - DC2 Server Rack 20,Asset Number: 10027087/1 / ,10-Sep-12,134,RP-SOI,Ready to Deploy,,NULL
,,,Ethernet Adapter for Server (IBM NetXtreme II 1000 PCI-E ),Ethernet Adapter,NetXtreme ,IBM,II PCI-E,X61S8TR,OIS24013,E5 - DC2 Server Rack 20,Asset Number: 10027087/2 / ,10-Sep-12,500.35,RP-SOI,Ready to Deploy,,NULL
,,,Ethernet Adapter for Server (IBM NetXtreme II 1000 PCI-E ),Ethernet Adapter,NetXtreme ,IBM,II PCI-E,X61S8T0,OIS24015,E5 - DC2 Server Rack 20,Asset Number: 10027088/1 / ,10-Sep-12,1687,RP-SOI,Ready to Deploy,,NULL
,,,Ethernet Adapter for Server (IBM NetXtreme II 1000 PCI-E ),Ethernet Adapter,NetXtreme ,IBM,II PCI-E,X61S8TW,OIS24016,E5 - DC2 Server Rack 20,Asset Number: 10027088/2 / ,10-Sep-12,134,RP-SOI,Ready to Deploy,,NULL

**Comment by dmeltzer** [comment]
Hello,

I'm wondering if my PR #2396 will fix this.  Php 7 on linux throws an exception after import, but it looks like there are some interpretters that will do exactly what this demonstrates (See https://3v4l.org/KuQ9l)

**Comment by CLOZADA16** [comment]
I followed what i see in [#2396](https://github.com/snipe/snipe-it/pull/2396/commits/5183cc1958af0d5e1d265ac46a16f43ff3042f2a) - changed the things highlighted in red to those highlighted in green. Tried importing the file after changing AssetsController.php but I'm still getting the same result.

**Comment by CLOZADA16** [comment]
Hi, I was wondering if a solution for this has been found?

**Comment by dmeltzer** [comment]
A number of changes have been merged to the `develop` branch relating to the importer.  If you have a chance, could you test this code and see if it fixes things for you?

**Comment by ZechClar** [comment]
Hi dmeltzer, this has been fixed after I upgraded to v3.6.2. Thank you
